readme.md
<div align="center">
  <p>
    <img alt="Logo" src="https://raw.githubusercontent.com/spear-ai/logo/main/src/assets/logo.svg" width="192">
  </p>
  <p>
    <b>Spear AI Logo</b>
  </p>
  <p>
    <a href="https://github.com/spear-ai/logo/actions/workflows/checks.yml">
      <img alt="checks" src="https://img.shields.io/github/workflow/status/spear-ai/logo/checks?labelColor=3A3a40&color=04C835&label=checks&logo=github">
    </a>
    <a href="https://conventionalcommits.org">
      <img alt="Conventional Commits" src="https://img.shields.io/badge/Conventional_Commits-1.0.0-ff206E.svg?labelColor=3A3a40">
    </a>
  </p>
</div>


changelog.md
# Changelog

## [2.1.1](https://github.com/spear-ai/logo/compare/logo-v2.1.0...logo-v2.1.1) (2023-07-30)


### Bug Fixes

* **logo:** fix logo shape ([#169](https://github.com/spear-ai/logo/issues/169)) ([4d86e2b](https://github.com/spear-ai/logo/commit/4d86e2b9a708108b15fb5b8375ebfa714a13d06b))

## [2.1.0](https://github.com/spear-ai/logo/compare/logo-v2.0.27...logo-v2.1.0) (2022-05-16)


### Features

* add logo text! ([#98](https://github.com/spear-ai/logo/issues/98)) ([521f069](https://github.com/spear-ai/logo/commit/521f069bffc3f8c4294d0a4e78912c84208fefde))

### [2.0.27](https://github.com/spear-ai/logo/compare/logo-v2.0.26...logo-v2.0.27) (2022-05-09)


### Bug Fixes

* fix gpr ([#96](https://github.com/spear-ai/logo/issues/96)) ([5bf4682](https://github.com/spear-ai/logo/commit/5bf46821c86a69995c945eaf295caed48f0feeca))

### [2.0.26](https://github.com/spear-ai/logo/compare/logo-v2.0.25...logo-v2.0.26) (2022-05-06)


### Bug Fixes

* fix publish to GitHub packages ([d076a5b](https://github.com/spear-ai/logo/commit/d076a5b5a298a853377de0322a8526b06a80d56e))

### [2.0.25](https://github.com/spear-ai/logo/compare/logo-v2.0.24...logo-v2.0.25) (2022-05-06)


### Bug Fixes

* publish to both NPM and GitHub Packages ([#91](https://github.com/spear-ai/logo/issues/91)) ([c51c258](https://github.com/spear-ai/logo/commit/c51c258054b86e6c2cf35679bf202fded405cd2a))

### [2.0.24](https://github.com/spear-ai/logo/compare/logo-v2.0.23...logo-v2.0.24) (2022-05-05)


### Bug Fixes

* fix release ([#89](https://github.com/spear-ai/logo/issues/89)) ([b21b207](https://github.com/spear-ai/logo/commit/b21b2078ae8d3312a4504c404c0b5b3c04fcc475))

### [2.0.23](https://github.com/spear-ai/logo/compare/logo-v2.0.22...logo-v2.0.23) (2022-03-19)


### Bug Fixes

* fix cjs exports ([a77076c](https://github.com/spear-ai/logo/commit/a77076c43f8c290a4fa5fd485ca8015198e4bfd5))

### [2.0.22](https://github.com/spear-ai/logo/compare/logo-v2.0.21...logo-v2.0.22) (2022-03-18)


### Bug Fixes

* fix publish ([06d1163](https://github.com/spear-ai/logo/commit/06d1163d62c10287d8c10a17788dbb5c11477543))

### [2.0.21](https://github.com/spear-ai/logo/compare/logo-v2.0.20...logo-v2.0.21) (2022-03-07)


### Bug Fixes

* fix publish ([32c46f9](https://github.com/spear-ai/logo/commit/32c46f90ce7297f1b7d39a6d2de0eed0f96c65ce))

### [2.0.20](https://github.com/spear-ai/logo/compare/logo-v2.0.19...logo-v2.0.20) (2022-03-07)


### Bug Fixes

* fix publish ([969dce1](https://github.com/spear-ai/logo/commit/969dce19ed7d5684b2d08cc05a31e3f31829ddc8))

### [2.0.19](https://github.com/spear-ai/logo/compare/logo-v2.0.18...logo-v2.0.19) (2022-03-07)


### Bug Fixes

* fix publish ([735aa41](https://github.com/spear-ai/logo/commit/735aa41843c74afa1ff442b840b27b7e5961654b))

### [2.0.18](https://github.com/spear-ai/logo/compare/logo-v2.0.17...logo-v2.0.18) (2022-03-07)


### Bug Fixes

* fix publish ([3814269](https://github.com/spear-ai/logo/commit/381426991a9951088ab587216816e59ec2940312))

### [2.0.17](https://github.com/spear-ai/logo/compare/logo-v2.0.16...logo-v2.0.17) (2022-03-07)


### Bug Fixes

* fix publish ([0561b28](https://github.com/spear-ai/logo/commit/0561b28f88b4eace160b4e4acedfd1dad2ce4948))

### [2.0.16](https://github.com/spear-ai/logo/compare/logo-v2.0.15...logo-v2.0.16) (2022-03-07)


### Bug Fixes

* fix publish ([e009f20](https://github.com/spear-ai/logo/commit/e009f2086a0b69f29ce958a34d75584528228305))

### [2.0.15](https://github.com/spear-ai/logo/compare/logo-v2.0.14...logo-v2.0.15) (2022-03-07)


### Bug Fixes

* fix publish ([4f03aee](https://github.com/spear-ai/logo/commit/4f03aee609dc5d495c463e2db96387e6812476a0))

### [2.0.14](https://github.com/spear-ai/logo/compare/logo-v2.0.13...logo-v2.0.14) (2022-03-07)


### Bug Fixes

* fix publish ([93e0e7e](https://github.com/spear-ai/logo/commit/93e0e7ec7e011f04b69741ba5827e92b0811e33b))

### [2.0.13](https://github.com/spear-ai/logo/compare/logo-v2.0.12...logo-v2.0.13) (2022-03-07)


### Bug Fixes

* fix publish ([23c47d6](https://github.com/spear-ai/logo/commit/23c47d6602b54c449714712366b87e009a1ab689))

### [2.0.12](https://github.com/spear-ai/logo/compare/logo-v2.0.11...logo-v2.0.12) (2022-03-07)


### Bug Fixes

* fix publish ([3975e4a](https://github.com/spear-ai/logo/commit/3975e4aa7b09aea5e728ed2c2ac1277e741af109))

### [2.0.11](https://github.com/spear-ai/logo/compare/logo-v2.0.10...logo-v2.0.11) (2022-03-07)


### Bug Fixes

* fix publish ([5ee86fe](https://github.com/spear-ai/logo/commit/5ee86fe9ee6221d456572a0827776a1bdceea1c9))

### [2.0.10](https://github.com/spear-ai/logo/compare/logo-v2.0.9...logo-v2.0.10) (2022-03-07)


### Bug Fixes

* fix publish ([6ebfc00](https://github.com/spear-ai/logo/commit/6ebfc00153381c8f0472cbd28a96c948e66088dc))

### [2.0.9](https://github.com/spear-ai/logo/compare/logo-v2.0.8...logo-v2.0.9) (2022-03-07)


### Bug Fixes

* fix publish ([d2e71d7](https://github.com/spear-ai/logo/commit/d2e71d73cc5f35a4d5ed77eab3f06f373195c547))

### [2.0.8](https://github.com/spear-ai/logo/compare/logo-v2.0.7...logo-v2.0.8) (2022-03-07)


### Bug Fixes

* fix publish ([9ac8386](https://github.com/spear-ai/logo/commit/9ac838678e5d21971acb80e7b04834283255ab5b))

### [2.0.7](https://github.com/spear-ai/logo/compare/logo-v2.0.6...logo-v2.0.7) (2022-03-07)


### Bug Fixes

* fix publish ([6fa2040](https://github.com/spear-ai/logo/commit/6fa2040d93adfeb8858733a4b06a446916e5543a))

### [2.0.6](https://github.com/spear-ai/logo/compare/logo-v2.0.5...logo-v2.0.6) (2022-03-07)


### Bug Fixes

* fix publish ([733d66e](https://github.com/spear-ai/logo/commit/733d66e9992bd01c6f10d6378307e5778ff58f25))

### [2.0.5](https://github.com/spear-ai/logo/compare/logo-v2.0.4...logo-v2.0.5) (2022-03-07)


### Bug Fixes

* fix publish ([c4c4834](https://github.com/spear-ai/logo/commit/c4c4834c476e31d3ad5c4e24c743ef00f8782299))

### [2.0.4](https://github.com/spear-ai/logo/compare/logo-v2.0.3...logo-v2.0.4) (2022-03-07)


### Bug Fixes


src/react/path-builder.js
import fs from 'node:fs';

const x = 50; // This scale factor was probably not needed, as it is a vector graphic.
// n: This is the important scale factor. Geometrically, it is the main side length of the
// inner irregular hexagon. If you normalize the length of the "chopped off" sides to
// length of 1, then this is the length of the longer sides.
const n = 3.33; // 3.33 is the magic value to make this look just like the original logo
const w = (8 + n) * x;
const h = (w / 2) * Math.sqrt(3);

// Really big string template
const svg = `<svg
  viewBox="${1.5 * x} ${1.5 * Math.sqrt(3) * x} ${w - 3 * x} ${h - 1.5 * Math.sqrt(3) * x}"
  xmlns="http://www.w3.org/2000/svg"
>
  <style>
  @media (prefers-color-scheme: dark) {
    .primary {
      fill: #3070b8;
    }
    .secondary {
      fill: #fafaff;
    }
  }
  </style>
  <defs>
    <path class="primary" id="logo_arc" d="
M 0,${h}
m ${3 * x},0
l ${x},0
l ${-1.5 * x}, ${-1.5 * Math.sqrt(3) * x}
l ${(n * x) / 2}, ${-(n * x * Math.sqrt(3)) / 2}
l ${-x},0
l ${-(n * x) / 2}, ${(n * x * Math.sqrt(3)) / 2}
Z"
      fill="#2f6fb6"
    />
  </defs>
  <use href="#logo_arc" />
  <use href="#logo_arc" 
    transform="translate(${(3.5 + n) * x},${-1.5 * x * Math.sqrt(3)})
    rotate(-120,${3 * x},${h})"
  />
  <use href="#logo_arc" 
    transform="translate(${((n - 1) / 2) * x},${-h + 1.5 * Math.sqrt(3) * x})
    rotate(120,${3 * x},${h})"
  />
  <polygon class="secondary" points="
    ${4 * x},${h - Math.sqrt(3) * x}
    ${(4 + n) * x},${h - Math.sqrt(3) * x}
    ${(4.5 + n) * x},${h - (1.5 * Math.sqrt(3) * x)}
    ${(4.5 + n / 2) * x},${(2.5 * Math.sqrt(3)) * x}
    ${(3.5 + n / 2) * x},${(2.5 * Math.sqrt(3)) * x}
    ${3.5 * x},${h - (1.5 * Math.sqrt(3) * x)}
    "
    fill="#000031"
  />
</svg>`;

// Write the svg to file
fs.writeFile('src/assets/better-logo.svg', svg, () => {});


src/build-logo-icon.js
import fs from 'node:fs/promises';

const scale = 50; // This scale factor was probably not needed, as it is a vector graphic.
// n: This is the important scale factor. Geometrically, it is the main side length of the
// inner irregular hexagon. If you normalize the length of the "chopped off" sides to
// length of 1, then this is the length of the longer sides.
const ratio = 3.33; // 3.33 is the value to make this look just like the original logo
const width = (8 + ratio) * scale;
const height = (width / 2) * Math.sqrt(3);

// Really big string template
const svg = `<svg width="400" height="400"
  viewBox="${1.5 * scale} ${1.5 * Math.sqrt(3) * scale} ${width - 3 * scale} ${height - 1.5 * Math.sqrt(3) * scale}"
  xmlns="http://www.w3.org/2000/svg"
>
  <style>
  @media (prefers-color-scheme: dark) {
    .primary {
      fill: #3070b8;
    }
    .secondary {
      fill: #fafaff;
    }
  }
  </style>
  <defs>
    <path class="primary" id="logo_arc" d="
M 0,${height}
m ${3 * scale},0
l ${scale},0
l ${-1.5 * scale}, ${-1.5 * Math.sqrt(3) * scale}
l ${(ratio * scale) / 2}, ${-(ratio * scale * Math.sqrt(3)) / 2}
l ${-scale},0
l ${-(ratio * scale) / 2}, ${(ratio * scale * Math.sqrt(3)) / 2}
Z"
      fill="#2f6fb6"
    />
  </defs>
  <use href="#logo_arc" />
  <use href="#logo_arc" 
    transform="translate(${(3.5 + ratio) * scale},${-1.5 * scale * Math.sqrt(3)})
    rotate(-120,${3 * scale},${height})"
  />
  <use href="#logo_arc" 
    transform="translate(${((ratio - 1) / 2) * scale},${-height + 1.5 * Math.sqrt(3) * scale})
    rotate(120,${3 * scale},${height})"
  />
  <polygon class="secondary" points="
    ${4 * scale},${height - Math.sqrt(3) * scale}
    ${(4 + ratio) * scale},${height - Math.sqrt(3) * scale}
    ${(4.5 + ratio) * scale},${height - (1.5 * Math.sqrt(3) * scale)}
    ${(4.5 + ratio / 2) * scale},${(2.5 * Math.sqrt(3)) * scale}
    ${(3.5 + ratio / 2) * scale},${(2.5 * Math.sqrt(3)) * scale}
    ${3.5 * scale},${height - (1.5 * Math.sqrt(3) * scale)}
    "
    fill="#000031"
  />
</svg>`;

await fs.writeFile('src/assets/logo-icon.svg', svg);
await fs.writeFile('src/assets/favicon.svg', svg);
